Cannot find module `node_modules/@orama/plugin-data-persistence/dist/server-commonjs.cjs`

**Describe the bug**

Switching to `/server` exports per docs:

https://docs.oramasearch.com/plugins/plugin-data-persistence

```ts
import { persistToFile } from '@orama/plugin-data-persistence/server'
```

Getting the error:

```
Error: Cannot find module '.../node_modules/@orama/plugin-data-persistence/dist/server-commonjs.cjs'
```

Upon `dist` inspection, I don't see `server-commonjs.cjs` there (only `server-commonjs.js`).

```
❯ ls node_modules/@orama/plugin-data-persistence/dist/   
      
commonjs.cjs           errors.js              index.js.map           server.d.ts            types.js               utils.js.map
commonjs.cjs.map       errors.js.map          server-commonjs.d.cts  server.js              types.js.map
commonjs.d.cts         index.d.ts             server-commonjs.js     server.js.map          utils.d.ts
errors.d.ts            index.js               server-commonjs.js.map types.d.ts             utils.js
```

**To Reproduce**

1. Install the module.
2. Try to import as documented.

**Expected behavior**

Imports should work as documented.

**Screenshots**

N/A

**Desktop (please complete the following information):**

N/A

**Smartphone (please complete the following information):**

N/A

**Additional context**

Applying the following patch fixes the runtime, but TypeScript type check still fails:

```patch
diff --git a/package.json b/package.json

--- a/package.json
+++ b/package.json
@@ -15,8 +15,8 @@
     "./server": {
       "types": "./dist/server.d.ts",
       "import": "./dist/server.js",
-      "require": "./dist/server-commonjs.cjs",
-      "node": "./dist/server-commonjs.cjs"
+      "require": "./dist/server-commonjs.js",
+      "node": "./dist/server-commonjs.js"
     }
   },
   "types": "./dist/index.d.ts",

```

--- a/.gitignore
+++ b/.gitignore
@@ -2,10 +2,11 @@
 .vscode/
 coverage/
 dist/
+tmp/
+**/tmp/
 node_modules/
 packages/plugin-docusaurus-sandbox/build
 playwright-report
 .dev
 .DS_Store
 .turbo
-rnr-v0.4.2-x86_64-unknown-linux-musl

--- a/packages/plugin-astro/package.json
+++ b/packages/plugin-astro/package.json
@@ -36,7 +36,7 @@
     "dev": "swc --delete-dir-on-start -s -w --extensions .ts -d dist src",
     "build": "swc --delete-dir-on-start --extensions .ts -d dist src",
     "postbuild": "tsc -p . --emitDeclarationOnly",
-    "pretest": "node ../../scripts/test-pack.mjs",
+    "pretest": "node ../../scripts/test-pack.mjs plugin-astro",
     "test": "node --loader=tsx --no-warnings=loader --test ./test/integration.ts | tap-mocha-reporter spec",
     "lint": "eslint src --ext .js,.ts,.tsx,.cts"
   },

--- a/packages/plugin-astro/test/integration.ts
+++ b/packages/plugin-astro/test/integration.ts
@@ -17,7 +17,7 @@ interface Execution {
 }
 
 const sandboxSource = fileURLToPath(new URL('./sandbox', import.meta.url))
-const sandbox = process.env.KEEP_SANDBOX
+const sandbox = process.env.KEEP_SANDBOX_ASTRO
   ? '/tmp/orama-astro-sandbox'
   : resolve(tmpdir(), `orama-plugin-astro-${Date.now()}`)
 
@@ -133,6 +133,6 @@ await test('generated DBs have indexed pages content', async () => {
   assert.ok(dynamicTestMissingSearchResult.count === 0)
 })
 
-if (!process.env.KEEP_SANDBOX) {
+if (!process.env.KEEP_SANDBOX_ASTRO) {
   await cleanup()
 }

--- a/packages/plugin-astro/test/sandbox/package.json
+++ b/packages/plugin-astro/test/sandbox/package.json
@@ -13,8 +13,8 @@
   },
   "pnpm": {
     "overrides": {
-      "@orama/orama": "@ROOT@/dist/orama-orama-@VERSION@.tgz",
-      "@orama/plugin-astro": "@ROOT@/dist/orama-plugin-astro-@VERSION@.tgz"
+      "@orama/orama": "@ROOT@/packages/plugin-astro/tmp/orama-orama-@VERSION@.tgz",
+      "@orama/plugin-astro": "@ROOT@/packages/plugin-astro/tmp/orama-plugin-astro-@VERSION@.tgz"
     }
   }
 }

--- a/packages/plugin-data-persistence/package.json
+++ b/packages/plugin-data-persistence/package.json
@@ -26,7 +26,7 @@
   "scripts": {
     "dev": "swc --delete-dir-on-start -s -w --extensions .ts,.cts -d dist src",
     "build": "swc --delete-dir-on-start --extensions .ts,.cts -d dist src",
-    "postbuild": "tsc -p . --emitDeclarationOnly && tsc -p tsconfig.cjs.json --emitDeclarationOnly && mv dist/commonjs.js dist/commonjs.cjs && mv dist/commonjs.js.map dist/commonjs.cjs.map",
+    "postbuild": "sh scripts/postbuild.sh",
     "test": "c8 -c test/config/c8.json tap --rcfile=test/config/tap.yml test/*.test.ts"
   },
   "keywords": [

--- a/packages/plugin-data-persistence/scripts/postbuild.sh
+++ b/packages/plugin-data-persistence/scripts/postbuild.sh
@@ -1,0 +1,12 @@
+#!/bin/bash
+
+set -x -e
+
+tsc -p . --emitDeclarationOnly
+tsc -p tsconfig.cjs.json --emitDeclarationOnly
+
+mv dist/commonjs.js dist/commonjs.cjs
+mv dist/commonjs.js.map dist/commonjs.cjs.map
+mv dist/server-commonjs.js dist/server-commonjs.cjs
+mv dist/server-commonjs.js.map dist/server-commonjs.cjs.map
+

--- a/packages/plugin-docusaurus/package.json
+++ b/packages/plugin-docusaurus/package.json
@@ -38,7 +38,7 @@
     "prebuild": "npm run lint",
     "build": "swc --delete-dir-on-start --extensions .ts,.tsx,.cts -d dist src",
     "postbuild": "sh scripts/postbuild.sh",
-    "pretest": "node ../../scripts/test-pack.mjs",
+    "pretest": "node ../../scripts/test-pack.mjs plugin-docusaurus",
     "test": "node --loader=tsx --no-warnings=loader --test ./test/integration.ts | tap-mocha-reporter spec",
     "format": "prettier -w src",
     "lint": "eslint src --ext .js,.ts,.tsx,.cts",

--- a/packages/plugin-docusaurus/test/integration.ts
+++ b/packages/plugin-docusaurus/test/integration.ts
@@ -20,7 +20,7 @@ interface Execution {
 }
 
 const sandboxSource = fileURLToPath(new URL('./sandbox', import.meta.url))
-const sandbox = process.env.KEEP_SANDBOX
+const sandbox = process.env.KEEP_SANDBOX_DOCUSAURUS
   ? '/tmp/orama-docusaurus-sandbox'
   : resolve(tmpdir(), `orama-plugin-docusaurus-${Date.now()}`)
 
@@ -121,6 +121,6 @@ await test('generated DBs have indexed pages content', async () => {
   assert.ok(turtleSearchResult.count === 0)
 })
 
-if (!process.env.KEEP_SANDBOX) {
+if (!process.env.KEEP_SANDBOX_DOCUSAURUS) {
   await cleanup()
 }

--- a/packages/plugin-docusaurus/test/sandbox/package.json
+++ b/packages/plugin-docusaurus/test/sandbox/package.json
@@ -18,10 +18,10 @@
   },
   "pnpm": {
     "overrides": {
-      "@orama/orama": "@ROOT@/dist/orama-orama-@VERSION@.tgz",
-      "@orama/plugin-docusaurus": "@ROOT@/dist/orama-plugin-docusaurus-@VERSION@.tgz",
-      "@orama/plugin-match-highlight": "@ROOT@/dist/orama-plugin-match-highlight-@VERSION@.tgz",
-      "@orama/plugin-parsedoc": "@ROOT@/dist/orama-plugin-parsedoc-@VERSION@.tgz"
+      "@orama/orama": "@ROOT@/packages/plugin-docusaurus/tmp/orama-orama-@VERSION@.tgz",
+      "@orama/plugin-docusaurus": "@ROOT@/packages/plugin-docusaurus/tmp/orama-plugin-docusaurus-@VERSION@.tgz",
+      "@orama/plugin-match-highlight": "@ROOT@/packages/plugin-docusaurus/tmp/orama-plugin-match-highlight-@VERSION@.tgz",
+      "@orama/plugin-parsedoc": "@ROOT@/packages/plugin-docusaurus/tmp/orama-plugin-parsedoc-@VERSION@.tgz"
     },
     "packageExtensions": {
       "@docusaurus/theme-live-codeblock": {

--- a/scripts/test-pack.mjs
+++ b/scripts/test-pack.mjs
@@ -1,11 +1,12 @@
 import assert from 'node:assert'
 import { exec } from 'node:child_process'
 import { existsSync } from 'node:fs'
-import { readFile } from 'node:fs/promises'
+import { readFile, rm } from 'node:fs/promises'
 import { resolve } from 'node:path'
 import { fileURLToPath } from 'node:url'
 
 const rootDir = fileURLToPath(new URL('../', import.meta.url))
+const destinationDir = resolve(rootDir, 'packages', process.argv[2], 'tmp')
 
 async function execute(command, cwd) {
   const { HOME, PATH } = process.env
@@ -31,8 +32,11 @@ async function execute(command, cwd) {
   })
 }
 
-const pluginInfo = JSON.parse(await readFile(fileURLToPath(new URL('../package.json', import.meta.url)), 'utf-8'))
+const pluginInfo = JSON.parse(
+  await readFile(fileURLToPath(new URL('../packages/orama/package.json', import.meta.url)), 'utf-8'),
+)
 
+await rm(destinationDir, { force: true, recursive: true })
 for (const pkg of ['orama', 'plugin-match-highlight', 'plugin-parsedoc', 'plugin-astro', 'plugin-docusaurus']) {
   const pluginPath = fileURLToPath(new URL(`../dist/orama-${pkg}-${pluginInfo.version}.tgz`, import.meta.url))
 
@@ -41,7 +45,7 @@ for (const pkg of ['orama', 'plugin-match-highlight', 'plugin-parsedoc', 'plugin
   }
 
   console.log(`\x1b[1m\x1b[32m--- Packing @orama/${pkg}-${pluginInfo.version} for testing ...\x1b[0m`)
-  const packResult = await execute('pnpm pack --pack-destination ../../dist', resolve(rootDir, 'packages', pkg))
+  const packResult = await execute(`pnpm pack --pack-destination ${destinationDir}`, resolve(rootDir, 'packages', pkg))
   console.log(packResult.stdout.trim())
   assert.equal(packResult.code, 0)
 }

--- a/scripts/version.mjs
+++ b/scripts/version.mjs
@@ -4,6 +4,8 @@ import { simpleGit } from 'simple-git'
 import { fileURLToPath } from 'url'
 import oramaPackage from '../packages/orama/package.json' assert { type: 'json' }
 
+const dryRun = process.env.DRY_RUN === 'true'
+const skipGit = process.env.SKIP_GIT === 'true'
 const skip = (process.env.SKIP ?? '').split(/\s*,\s*/).map(s => s.trim())
 
 const packages = [
@@ -12,8 +14,24 @@ const packages = [
   'plugin-data-persistence',
   'plugin-docusaurus',
   'plugin-match-highlight',
+  'plugin-nextra',
   'plugin-parsedoc',
+  'stemmers',
+  'stopwords',
 ].filter(p => !skip.includes(p))
+
+async function updatePackage(path, version) {
+  const packageJson = JSON.parse(await readFile(path))
+
+  console.log(
+    `\x1b[32mUpdating package \x1b[1m${packageJson.name}\x1b[22m from \x1b[1m${packageJson.version}\x1b[22m to version \x1b[1m${version}\x1b[22m ...\x1b[0m`,
+  )
+  packageJson.version = version
+
+  if (!dryRun) {
+    await writeFile(path, JSON.stringify(packageJson, null, 2), 'utf-8')
+  }
+}
 
 async function main() {
   if (!process.argv[2]) {
@@ -35,25 +53,16 @@ async function main() {
     process.exit(1)
   }
 
+  // Update the main package.json
+  await updatePackage(fileURLToPath(new URL(`../package.json`, import.meta.url)), version)
+
   // Perform the update
   for (const pkg of packages) {
-    const packageJsonPath = fileURLToPath(new URL(`../packages/${pkg}/package.json`, import.meta.url))
-    const packageJson = JSON.parse(await readFile(packageJsonPath))
-
-    console.log(
-      `\x1b[32mUpdating package \x1b[1m${packageJson.name}\x1b[22m from \x1b[1m${packageJson.version}\x1b[22m to version \x1b[1m${version}\x1b[22m ...\x1b[0m`,
-    )
-    packageJson.version = version
-
-    if (process.env.DRY_RUN) {
-      continue
-    }
-
-    await writeFile(packageJsonPath, JSON.stringify(packageJson, null, 2), 'utf-8')
+    await updatePackage(fileURLToPath(new URL(`../packages/${pkg}/package.json`, import.meta.url)), version)
   }
 
   // Make the commit, including the tag
-  if (!process.env.DRY_RUN) {
+  if (!dryRun && !skipGit) {
     const git = simpleGit({ baseDir: fileURLToPath(new URL('..', import.meta.url)) })
     console.log(`\x1b[32mCommitting changes and creating a tag (pushing is left to you) ...\x1b[0m`)
     await git.commit(

